hathor/transaction/storage/transaction_storage.py

```diff
@@ -219,6 +219,15 @@ def get_rev_deps(self, tx: bytes) -> FrozenSet[bytes]:
         """Get all txs that depend on the given tx (i.e. its reverse depdendencies)."""
         return frozenset(self._rev_dep_index.get(tx, set()))
 
+    def children_from_deps(self, tx: bytes) -> List[bytes]:
+        """Return the hashes of all reverse dependencies that are children of the given tx.
+
+        That is, they depend on `tx` because they are children of `tx`, and not because `tx` is an input. This is
+        useful for pre-filling the children metadata, which would otherwise only be updated when
+        `update_initial_metadata` is called on the child-tx.
+        """
+        return [not_none(rev.hash) for rev in map(self.get_transaction, self.get_rev_deps(tx)) if tx in rev.parents]
```

Comment:
As far as I understood, the `if tx in rev.parents` could be an assert, right? I mean, it should be impossible to fail except for a bug in the rev deps index, right?

Reply:
A tx depends on its parent _and inputs_, which will make `self.get_rev_deps(tx)` also return the transactions that have `tx` as input and might not have it on its parents.

Reply:
Old conversation. It does make sense anymore. Marking as resolved.